### PR TITLE
Style Hints According to Config

### PR DESCRIPTION
**Pull request by Chic-Tweetz**
As discussed in https://github.com/tridactyl/tridactyl/issues/4106 and others, hint element styles can't be completely disabled in themes.
This adds a config key, `hintstyles` so that styles are only added if they are enabled, otherwise the original page style remains intact.
Outlines, backgrounds and font colour can individually be enabled/disabled, or only added to active hint using `set hintstyles.bg`, `hintstyles.outline` and `hintstyles.fg`. Values can be `all`, `active`, or `none`

### CODE REVIEW

**Comment by bovine3dom** [comment]
this is a neat solution, thanks! I need to find some time to check through it to make sure e.g. it isn't breaking people's own custom themes as far as possible. probably this weekend.

**Comment by Chic-Tweetz** [comment]
On that point, all three default settings should be "all" in config.ts rather than "active" to keep themes as they are now.

File: src/lib/config.ts [hunk 1]
```diff
@@ -1334,6 +1334,15 @@ export class default_config {
      * Whether to show article url in the document.title of Reader View.
      */
     readerurlintitle: "true" | "false" = "false"
+
+    /**
+     *  Which css styles to add for hint elements.
+     **/
+    hintstyles: { [key: string]: "all" | "active" | "none" } = {
+        fg: "active",
+        bg: "active",
```
Comment (bovine3dom) [inline src/lib/config.ts:1343]:
```suggestion
        fg: "all",
        bg: "all",
```

File: src/lib/config.ts [hunk 2]
```diff
@@ -1334,6 +1334,15 @@ export class default_config {
      * Whether to show article url in the document.title of Reader View.
      */
     readerurlintitle: "true" | "false" = "false"
+
+    /**
+     *  Which css styles to add for hint elements.
```
Comment (bovine3dom) [inline src/lib/config.ts:1339]: We should say what fg/bg/outline mean and give an example of how to set it, e.g. `:set hintstyles.bg none`

**Review by bovine3dom — COMMENTED** [review]
Thanks, just a couple of things then, happy to change them myself when I get a proper chance to look at this this weekend

**Pull request merged**